src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs

```diff
@@ -832,14 +838,20 @@ private static FormulaValue LeftOrRight(IRContext irContext, FormulaValue[] args
                 throw new NotImplementedException("Should have been handled by IR");
             }
 
-            return new StringValue(irContext, leftOrRight(source.Value, (int)count.Value));
+            TryGetIntValue(count.Value, out int cnt);
```

Comment:
>TryGetIntValue [](http://example.com/codeflow?start=12&length=14)

check return?

Reply:
we do not need to check failure here because if it is out of bound, we got the min or max of int.